kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -613,6 +604,46 @@ public LogSegment getLogSegmentForVersion(
   // getLogSegment utils //
   /////////////////////////
 
+  private List<ParsedLogData> getAllDeltasAfterCheckpointWithCatalogPriority(
+      List<ParsedLogData> allPublishedDeltas,
+      List<ParsedLogData> parsedLogDatas,
```

Comment:
Do we enforce that parsedLogDatas is continuous, and sorted by version?

Reply:
Maybe consider dropping a note here as it is not obvious if I just read this piece of code?